### ISSUE TITLE
Misleading SFT baseline result on GSM8K.

### ISSUE THREAD

**Issue by WJMacro** [Julia-LiuJ/NLFT#4]
Thanks for releasing the code. 

After reading the paper, I found the SFT baseline is unusually low. I check the code, finding that for SFT the authors utilized the **original answer** of GSM8K dataset as the reference response (see line 211-212 in `sft.py`). However, in the proposed NLFT, the reference response is the **COT result of Llama-3-8B-Instruct** (see line 509,523 in `finetune_unlikelihood_dynamic_correct.py`), which achieves a zero-shot accuracy of over 80% on GSM8K. Therefore, the comparison between SFT and NLFT in the paper is unfair and potentially misleading, as the performance gain could come from **distilling the COT result** of Llama-3-8B-Instruct.

Utilizing the released NLFT800 dataset, I conducted the following experiments:

|                     | reference                                                                      | Acc    |
|---------------------|--------------------------------------------------------------------------------|--------|
| NLFT800-vanilla-ft  | 'answer' (ground truth of GSM8K)                                               | 51.40% |
| NLFT800-COT-kd      | 'output' (COT result of Llama-3-8B-Instruct)                                   | 69.60% |
| NLFT800-COT-cleaned | 'output' if the COT result is correct ('judgement'==null), otherwise 'answer'. | 69.29% |
| NLFT800-NLFT        | 'output'                                                                       | 67.8%  |

The same hyperparameter is utilized and I finetuned the model for 3 epochs. NLFT800-NLFT is the result reported in Figure.9 of the paper.

As shown in the table, **SFT with 'output' as the reference could achieve even higher result compared to NLFT**. I think these results demonstrates that in current experiments, performance gain mainly comes from distilling the COT result of Llama-3-8B-Instruct. 

Looking forward to more indepth discussion with the authors.

**Comment by Garen-Wang** [comment]
Thanks for your in-depth discussion. We are conducting ablation study to figure out the improvement of NLFT and mitigate impact of CoT from 8B-Instruct. We will update our results after holiday.

**Comment by chuangzhidan** [comment]
> ut' (COT result of Llama-3-8B-Instruct)	69.60

what about small sample size , for example ,50 instead of 800，sft(lora) is still comparable to NLFT？
